examples/showcase/eclipse_amount.py

```diff
@@ -0,0 +1,66 @@
+"""
+===================================
+Obtaining solar-eclipse information
+===================================
+
+How to obtain information about a solar eclipse
+
+Here we look into the total solar eclipse on 2024 April 8, which passes over
+Dallas, Texas.
+
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.dates import DateFormatter
+
+import astropy.units as u
+from astropy.coordinates import EarthLocation, solar_system_ephemeris
+from astropy.time import Time
+
+from sunpy.coordinates import sun
+
+##############################################################################
+# Create an array of observation times centered around the maximum of total
+# solar eclipse at ~18:43 UTC.  We sample per second within 5 minutes of
+# maximum, and per minute further out.
+
+obstime = (Time('2024-04-08 18:43')
+           + np.concatenate([np.arange(-100, -5) * u.min,
+                             np.arange(-300, 300) * u.s,
+                             np.arange(5, 100) * u.min]))
+
+##############################################################################
+# Create an observer coordinate for the longitude and latitude of Dallas for
+# the array of observation times.
+
+location = EarthLocation.from_geodetic(-96.808891*u.deg, 32.779167*u.deg)
+observer = location.get_itrs(obstime)
```

Comment:
Somewhat off on a tangent, but can we pass `EarthLocation` as an observer? i.e. skip this line. If not why not? :laughing:

Reply:
Because `EarthLocation` doesn't include `obstime`?

Reply:
*Drives the PR car off down another street*

Couldn't we still do the conversion though? Wont most transforms pull the obstime from the frame if it's missing on the observer?

Reply:
I think you're not looking at how the `observer` variable is later used.  It's not being used solely as the value for an `observer` frame attribute.  The function `eclipse_amount()` takes a single argument, which is the coordinate of the observer including the `obstime`.  If the function were to accept `EarthLocation` for the observer location, it would need to additionally require the `obstime` to be supplied as a second argument.